### PR TITLE
makeURL: Cleanup, test, and always default to `percent` space encoding

### LINKED ISSUE #3031: Default Open Tab Brick to percent encoding
- The current default is plus encoding, which is used for form encoding
- http://github.com/pixiebrix/pixiebrix-extension/blob/f0de53296859d6fedcde0660949faa1d433507ba/src/blocks/effects/redirectPage.ts#L48-L48

### PR DESCRIPTION
**Pull request by fregante**
- Closes #3031 

But I wonder why/when supporting both encodings would be beneficial. Even the native and modern `URL` constructor uses plus by default

### CODE REVIEW

File: src/blocks/effects/redirectPage.ts [hunk 1]
```diff
@@ -36,11 +33,7 @@ export class NavigateURLEffect extends Effect {
 
   inputSchema = URL_INPUT_SPEC;
 
-  async effect({
-    url,
-    params,
-    spaceEncoding = URL_INPUT_SPACE_ENCODING_DEFAULT,
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/blocks/effects/redirectPage.ts:42]: For backward compatibility (we don't want to break existing bricks that don't provide a value), we need this to be "plus". Should leave a comment that we're leaving it as plus for backward compatibility.

File: src/blocks/effects/redirectPage.ts [hunk 2]
```diff
@@ -57,11 +50,7 @@ export class OpenURLEffect extends Effect {
...
```
Comment (twschiller) [inline src/blocks/effects/redirectPage.ts:63]: Same comment as above on URL_INPUT_SPACE_ENCODING_DEFAULT

File: src/blocks/transformers/url.ts [hunk 3]
```diff
@@ -78,7 +76,7 @@ export class UrlParams extends Transformer {
   async transform({
     url,
     params,
-    spaceEncoding = URL_INPUT_SPACE_ENCODING_DEFAULT,
```
Comment (twschiller) [inline src/blocks/transformers/url.ts:81]: Same comment as for NavigateURLEffect

File: src/utils.test.ts [hunk 4]
```diff
@@ -152,3 +153,34 @@ describe("matchesAnyPattern", () => {
     expect(matchesAnyPattern("hello", [/^Hello$/])).toBeFalsy();
   });
 });
+
+describe("makeURL", () => {
```
Comment (twschiller) [inline src/utils.test.ts:157]: ❤️

**Comment by twschiller** [comment]
> But I wonder why/when supporting both encodings would be beneficial. Even the native and modern URL constructor uses plus by default

There's some idiosyncratic behavior in clients, especially when using custom URL schemes that open desktop applications. IIRC, MS Outlook appears to requires percent encoding?

My understanding percent encoding should always work, but plus encoding will only work if the client is OK with form encoded data? https://stackoverflow.com/a/2678602/402560

I might be missing something though? I find some of the terminology hard to keep track of

**Review by twschiller — CHANGES_REQUESTED** [review]
See comment on preserving the default in brick implementations for backward compatability

**Comment by fregante** [comment]
Let me know if I understood you correctly

File: src/blocks/effects/redirectPage.ts [hunk 5]
```diff
@@ -39,7 +36,7 @@ export class NavigateURLEffect extends Effect {
   async effect({
     url,
     params,
-    spaceEncoding = URL_INPUT_SPACE_ENCODING_DEFAULT,
+    spaceEncoding = LEGACY_URL_INPUT_SPACE_ENCODING_DEFAULT,
```
Comment (twschiller) [inline src/blocks/effects/redirectPage.ts:39]: 👍

File: src/blocks/effects/redirectPage.ts [hunk 6]
```diff
@@ -60,7 +57,7 @@ export class OpenURLEffect extends Effect {
...
```
Comment (twschiller) [inline src/blocks/effects/redirectPage.ts:60]: 👍

File: src/blocks/transformers/url.ts [hunk 7]
```diff
@@ -78,7 +80,7 @@ export class UrlParams extends Transformer {
...
+    spaceEncoding = LEGACY_URL_INPUT_SPACE_ENCODING_DEFAULT,
```
Comment (twschiller) [inline src/blocks/transformers/url.ts:83]: 👍

**Review by twschiller — APPROVED** [review]
image

**Pull request merged**